### PR TITLE
Resolved Minimum order Issue #202

### PR DESCRIPTION
**Pull request by rajlash**

### CODE REVIEW

**Comment by pramodshinde** [comment]
@rajlash Why test cases are failing?

File: app/controllers/admin_dashboard_controller.rb [hunk 1]
```diff
@@ -69,7 +69,25 @@ def confirm_orders
     Order.confirm_all_placed_orders(params[:terminal_id], current_user.company_id, @order_details)
     flash[:notice] = "all orders confirmed"
     redirect_to admin_dashboard_index_path
-  end 
+  end
+
+  def cancel_orders
```
Comment (pramodshinde) [inline app/controllers/admin_dashboard_controller.rb]: Have one method in model to cancel order and perform all operation in that model

File: app/controllers/admin_dashboard_controller.rb [hunk 2]
```diff
@@ -69,7 +69,25 @@ def confirm_orders
...
+      0
+    )
+    Order.cancel_all_placed_orders(
+      params[:terminal_id],
+      current_user.company_id,
+      @order_details
+    )
+    Company.where(id: current_user.company_id).
+            update(end_ordering_at: TIME_EXTENTION.minutes.from_now)
+    flash[:notice] = 'all orders cancelled'
```
Comment (pramodshinde) [inline app/controllers/admin_dashboard_controller.rb]: change message to `Orders has been cancelled`, Use locals for standard messages.

File: app/models/order.rb [hunk 3]
```diff
@@ -80,6 +80,25 @@ def self.confirm_all_placed_orders(terminal_id, company_id, order_details)
...
+    order_ids = order_details.pluck(:id).uniq
+    orders = Order.where(id: order_ids)
+    employee_ids =  orders.
+                    joins(:user).
+                    pluck('users.id')
+    recommended_terminals = Terminal.where(
+                              company_id: company_id,
+                              active:     true
+                            ).order(:min_order_amount)[0..2]
+    employee_ids.each do |employee_id|
```
Comment (pramodshinde) [inline app/models/order.rb]: Don't loop through employees, instead use array of email ids to send one email.

File: app/mailers/order_mailer.rb [hunk 4]
```diff
@@ -16,6 +16,24 @@ def send_mail_to_employees(employee)
     mail(to: email, subject: 'Status of Order')
   end
 
+  def send_order_cancel_employees(employee, recommended_terminals)
+    user = User.select(:name, :email).find(employee)
+    @name = user.name
```
Comment (pramodshinde) [inline app/mailers/order_mailer.rb]: Remove @name and email variables, instead use @user in template and in mail_to method

File: config/environments/production.rb [hunk 5]
```diff
@@ -23,7 +23,7 @@
   # config.assets.css_compressor = :sass
 
   # Do not fallback to assets pipeline if a precompiled asset is missed.
-  config.assets.compile = true
+#  config.assets.compile = true
```
Comment (pramodshinde) [inline config/environments/production.rb]: Remove this if these are not your changes

File: config/initializers/constants.rb [hunk 6]
```diff
@@ -1,4 +1,5 @@
 USER_ROLES= ['employee', 'company_admin']
 DAYS= ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday']
 ORDER_STATUS= ["pending", 'placed', 'confirmed']
-ORDER_DETAIL_STATUS= ['available', 'unavailable']
\ No newline at end of file
+ORDER_DETAIL_STATUS= ['available', 'unavailable']
+TIME_EXTENTION = 15 # Extra time window to order after cancel (in minutes)
```
Comment (pramodshinde) [inline config/initializers/constants.rb]: Add new line at the end.

File: db/seeds.rb [hunk 7]
```diff
@@ -1,48 +1,68 @@
+comp = Company.find_or_initialize_by(name: 'CompanyRaj', landline: '01234567890',email: 'raj@company.com',start_ordering_at: Time.zone.parse('12 AM'),end_ordering_at: Time.zone.parse('11 PM'),review_ordering_at: Time.zone.parse('11:30 PM'),subsidy: 50)
```
Comment (pramodshinde) [inline db/seeds.rb]: Remove seed changes.

File: public/Cristal Purdy I-invalid_records.csv [hunk 8]
```diff
@@ -0,0 +1,2 @@
+paneer,,true,best food,"1,2,3,4,5",Price can't be blank,Price is not a number
```
Comment (pramodshinde) [inline public/Cristal Purdy I-invalid_records.csv]: Remove all unnecessary files

File: test/controllers/admin_dashboard_controller_test.rb [hunk 9]
```diff
@@ -52,6 +51,27 @@ class AdminDashboardControllerTest  < ActionController::TestCase
     assert_redirected_to admin_dashboard_index_url
   end
 
+  test 'cancel order should extend ordering time window' do
+    sign_in_admin
+    create_orders
+    cancelling_time = Time.now
+    stub_request(:get, "http://hunger-terminal.s3.amazonaws.com/test/uploads/terminal/image/hotelplaceholder1.jpg")
```
Comment (pramodshinde) [inline test/controllers/admin_dashboard_controller_test.rb]: What is the response to stub request?

Comment (rajlash) [reply test/controllers/admin_dashboard_controller_test.rb]: Response is not needed here. Done.

**Pull request closed without merging**